StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1106CodeFixProvider.cs

```diff
@@ -106,20 +106,39 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
         private static async Task<Document> RemoveSemicolonTextAsync(Document document, SyntaxToken token, CancellationToken cancellationToken)
         {
+            TextChange textChange;
+
             SourceText sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
             TextLine line = sourceText.Lines.GetLineFromPosition(token.SpanStart);
             if (sourceText.ToString(line.Span).Trim() == token.Text)
             {
                 // remove the line containing the semicolon token
-                TextChange textChange = new TextChange(line.SpanIncludingLineBreak, string.Empty);
+                textChange = new TextChange(line.SpanIncludingLineBreak, string.Empty);
                 return document.WithText(sourceText.WithChanges(textChange));
             }
+
+            TextSpan spanToRemove;
+            var whitespaceIndex = TriviaHelper.IndexOfTrailingWhitespace(token.LeadingTrivia);
+            if (whitespaceIndex >= 0)
+            {
+                spanToRemove = TextSpan.FromBounds(token.LeadingTrivia[whitespaceIndex].Span.Start, token.Span.End);
```

Comment:
:memo: As of fae34cb9e28926b0ebba73e1bfa6ba2c7cad03e2, no test is covering this line.